### PR TITLE
8307779: Relax the java.awt.Robot specification

### PR DESCRIPTION
**Pull request by azvegint**
Please review the doc only backport PR of https://github.com/openjdk/jdk/pull/13809.

It is backport of the original commit + build failure fix.

CSR: [JDK-8331036](https://bugs.openjdk.org/browse/JDK-8331036)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8307779](https://bugs.openjdk.org/browse/JDK-8307779) needs maintainer approval
- [x] Change requires CSR request [JDK-8331036](https://bugs.openjdk.org/browse/JDK-8331036) to be approved

### Issues
 * [JDK-8307779](https://bugs.openjdk.org/browse/JDK-8307779): Relax the java.awt.Robot specification (**Bug** - P3 - Approved)
 * [JDK-8331036](https://bugs.openjdk.org/browse/JDK-8331036): Relax the java.awt.Robot specification (**CSR**)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-ri.git pull/10/head:pull/10` \
`$ git checkout pull/10`

Update a local copy of the PR: \
`$ git checkout pull/10` \
`$ git pull https://git.openjdk.org/jdk11u-ri.git pull/10/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10`

View PR using the GUI difftool: \
`$ git pr show -t 10`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-ri/pull/10.diff">https://git.openjdk.org/jdk11u-ri/pull/10.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-ri/pull/10#issuecomment-2075552327)

### CODE REVIEW

**Review by prrace — APPROVED** [review]

File: src/java.desktop/share/classes/java/awt/Robot.java [hunk 1]
```diff
@@ -389,11 +431,26 @@ private void checkKeycodeArgument(int keycode) {
...
+     *
      * @param   x       X position of pixel
      * @param   y       Y position of pixel
+     * @throws  SecurityException if {@code readDisplayPixels} permission
+     *          is not granted, or access to the screen is denied
+     *          by the desktop environment
      * @return  Color of the pixel
      */
     public synchronized Color getPixelColor(int x, int y) {
+        checkScreenCaptureAllowed();
```
Comment (irisclark) [inline src/java.desktop/share/classes/java/awt/Robot.java:453]: Note that the addition of this implementation change to comply with the spec will require that we produce an RI for the MR.

**Review by irisclark — APPROVED** [review]
Spec and implementation changes as expected for 11.0.0.1.

Associated CSR (JDK-8331036) already Approved.

**Review by gnu-andrew — APPROVED** [review]
Matches what we have for 17.0.0.1, bar a copyright header context difference and the need to add `checkScreenCaptureAllowed()`

**Pull request closed without merging**